lightning/src/util/config.rs

```diff
@@ -126,6 +126,26 @@ pub struct ChannelHandshakeConfig {
 	///
 	/// [`KeysInterface::get_shutdown_scriptpubkey`]: crate::chain::keysinterface::KeysInterface::get_shutdown_scriptpubkey
 	pub commit_upfront_shutdown_pubkey: bool,
+
+	/// `their_channel_reserve_satoshis` is the minimum amount that the other node is to keep as a direct payment.
+	/// This ensures that if our counterparty broadcasts a revoked state, we can punish them by
+	/// claiming at least this value on chain
+	///
+	/// `their_channel_reserve_proportion_millionth` is for us to configure counterparty's channel reserve
+	/// for both outgoing and incoming channels.
+	///
+	/// channel_reserve values greater than 30% could be considered highly unreasonable, since that
+	/// amount can never be used for payments.
+	/// Also, if our selected channel_reserve for counterparty and counterparty's selected channel_reserve
+	/// for us, sum upto or greater than channel_value, channel negotiations will fail.
+	///
+	/// `their_channel_reserve_proportion_millionth` is millionths of a channel_value to use as `their_channel_reserve_satoshis`.
+	///
+	/// Default Value is 1% of channel_value, configured as 10,000 millionth
+	/// Lower Bound on `their_channel_reserve_satoshis` is 1000 sats provided as safe implementation limit.
```

Comment:
I think this clearer should explain what happens if this value is set to a lower value than 1000, i.e.:
any values less than 1000 will be treated as 1000 instead.
This makes the config text more align with how other bounded values are explained (such as `cltv_expiry_delta` and `max_inbound_htlc_value_in_flight_percent_of_channel`).

Nit:
I also recommend formatting similar to how it's formatted in `cltv_expiry_delta` and `max_inbound_htlc_value_in_flight_percent_of_channel` i.e:
Default value: 10,000 ....
Minimum value: 1000 sats provided as ...
Maximum value: 1000,000 ....